amlb/data.py

```diff
@@ -35,16 +35,16 @@ def __init__(self, index, name, data_type, values=None, has_missing_values=False
         """
         self.index = index
         self.name = name
-        self.data_type = data_type
+        self.data_type = data_type.lower() if data_type is not None else None
         self.values = values
         self.has_missing_values = has_missing_values
         self.is_target = is_target
 
     def is_categorical(self, strict=True):
         if strict:
-            return self.data_type is not None and self.data_type.lower() in ['categorical', 'nominal', 'enum']
+            return self.data_type in ['categorical', 'nominal', 'enum']
```

Comment:
What about 'object' and 'str' types?

Reply:
we don't use 'object' type here: added a comment + creating a PR for those data types as they lack consistency and it's getting really confusing. Will probably try to use pandas DataFrame everywhere instead of numpy arrays.

For the current state of AutoGluon integration, I'll only convert numerical columns, the others will remain typed as `object`.
Is it fine for you?

Reply:
Seems fine to me, we don't do anything special beyond object, float, integer, and bool as input types.

Reply:
https://github.com/openml/automlbenchmark/issues/116